### PR TITLE
CLOUD-3918 - EAP 7.4.0.Beta OpenShift images

### PR DESCRIPTION
**Pull request by luck3y**
https://issues.redhat.com/browse/CLOUD-3918
Signed-off-by: Ken Wills <kwills@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by luck3y** [comment]
@jmesnil We've still got a build issue pending resolution, so this still points to some temporary branches. When I'm able to build locally, I'll update with the corrected tags & repos.

File: runtime-image/image.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      description: If set to true, ensures that the bash scripts are executed with the -x option, printing the commands and their arguments as they are executed.
+      example: "true"
+    - name: JBOSS_HOME
+      value: "/opt/eap"
+    # env var set by modules in the builder image context, duplicating them here.
+    - name: JBOSS_MODULES_SYSTEM_PKGS
+      value: "org.jboss.logmanager,jdk.nashorn.api"
+    - name: DEFAULT_ADMIN_USERNAME
+      value: "eapadmin"
+    - name: MICROPROFILE_CONFIG_DIR_ORDINAL
```
Comment (jmesnil) [inline runtime-image/image.yaml]: The EAP 7.4.0.Beta images no longer contain microprofile-config. This env var is not needed anymore

File: tests/features/basic.feature [hunk 2]
```diff
@@ -0,0 +1,14 @@
+@jboss-eap-7-tech-preview
+Feature: Common EAP tests
+
+  # https://issues.jboss.org/browse/CLOUD-180
+  Scenario: Check if image version and release is printed on boot
+    When container is ready
+    Then container log should contain Running jboss-eap-7/eap73-openjdk11-openshift-rhel8 image, version
```
Comment (jmesnil) [inline tests/features/basic.feature]: This should point to the EAP 7.4.0.Beta image, ,not the EAP 7.3 one.

File: tests/features/messaging.feature [hunk 3]
```diff
@@ -0,0 +1,11 @@
+@jboss-eap-7-tech-preview
+Feature: Openshift EAP Messaging Tests
+
+  Scenario: deploys the helloworld-mdb example, then checks if it's deployed properly.
+    Given s2i build https://github.com/jboss-developer/jboss-eap-quickstarts from helloworld-mdb with env using 7.2.0.GA
```
Comment (jmesnil) [inline tests/features/messaging.feature]: should we still use `env using 7.2.0.GA`?

File: image.yaml [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+          - name: jboss.container.wildfly.launch.access-log-valve
+          - name: jboss.container.wildfly.launch-config.config
+          - name: jboss.container.wildfly.launch-config.os
+          - name: jboss.container.wildfly.launch.datasources
+          - name: jboss.container.wildfly.launch.extensions
+          - name: jboss.container.wildfly.launch.json-logging
+          - name: jboss.container.wildfly.launch.jgroups
+          - name: jboss.container.wildfly.launch.filters
+          - name: jboss.container.wildfly.launch.logger-category
+          - name: jboss.container.wildfly.launch.mp-config
```
Comment (jmesnil) [inline image.yaml]: EAP 7.4.0.Beta no longer provides MicroProfile Config. This module should be removed

File: image.yaml [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+          - name: jboss.container.wildfly.launch.os.node-name
+          - name: jboss.container.wildfly.launch.tracing
```
Comment (jmesnil) [inline image.yaml]: EAP 7.4.0.Beta no longer provides MicroProfile OpenTracing. This module should be removed aiui

**Review by jmesnil — CHANGES_REQUESTED** [review]

File: image.yaml (hunk 4 above)
Comment (jfdenise) [reply image.yaml]: We also need to update the the jboss-eap-module to remove the scripts from the config: https://github.com/jboss-container-images/jboss-eap-modules/blob/master/jboss/container/eap/launch/added/launch/launch-config.sh#L31

File: image.yaml [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+          - name: wildfly-cekit-modules
+            git:
+                  url: https://github.com/luck3y/wildfly-cekit-modules.git
+                  ref: CLOUD-3918
+      install:
+          - name: jboss.container.openjdk.jdk
+            version: "11"
+          - name: jboss.container.maven
+            version: "8.2.3.6"
+          - name: eap-74-beta-env-latest
```
Comment (jfdenise) [inline image.yaml]: The correct env is: eap-74-beta-galleon-latest

Comment (luck3y) [reply image.yaml]: Thanks, actually this is what I had locally, i just hadn't pushed it, corrected.

**Review by jmesnil — APPROVED** [review]

**Pull request merged**